src/signed.rs

```diff
@@ -98,3 +119,59 @@ impl<'a, T: Signable, KB: KeyBox> From<Signed<'a, T, KB>> for UncheckedSigned<T,
         signed.into_unchecked()
     }
 }
+
+pub struct PartiallyMultisigned<'a, T: Signable, MK: MultiKeychain> {
+    unchecked: UncheckedSigned<T, MK::PartialMultisignature>,
+    indices: BoolNodeMap,
+    keychain: &'a MK,
+}
+
+pub struct Multisigned<'a, T: Signable, MK: MultiKeychain> {
+    pub unchecked: UncheckedSigned<T, MK::PartialMultisignature>,
+    pub indices: BoolNodeMap,
+    pub keychain: &'a MK,
+}
+
+pub struct IncompleteMultisignatureError<'a, T: Signable, MK: MultiKeychain> {
+    pub partial: PartiallyMultisigned<'a, T, MK>,
+}
+
+impl<'a, T: Signable, MK: MultiKeychain> PartiallyMultisigned<'a, T, MK> {
+    pub fn add_signature(&mut self, signature: &MK::Signature, index: NodeIndex) {
```

Comment:
There is not much code here but in a sense a lot happens here and a small mistake can have serious implications. Could you add a test? I understand that it requires a mock of `MultiKeychain` but the mock will be helpful in other tests as well.

Reply:
Added a test and fixed a mistake :)